src/beast/site_scons/site_tools/VSProject.py

```diff
@@ -583,6 +586,43 @@ def _guid(seed, name = None):
     guid = "{%s-%s-%s-%s-%s}" % (d[:8], d[8:12], d[12:16], d[16:20], d[20:32])
     return guid
 
+def _rmExeMod(file_name):
+    if platform.system() != 'Windows':
+        st = os.stat(file_name)
+        os.chmod(file_name, st.st_mode & ~0111)
+        return
+    
+    import win32security
+    import ntsecuritycon
+
+    user, domain, type = win32security.LookupAccountName ("", "Everyone")
+
+    sd = win32security.GetFileSecurity(file_name, win32security.DACL_SECURITY_INFORMATION)
+    dacl = sd.GetSecurityDescriptorDacl()
+
+    deniedAccess = ntsecuritycon.FILE_EXECUTE
+    dacl.AddAccessDeniedAce(win32security.ACL_REVISION, deniedAccess, user)
+
+    sd.SetSecurityDescriptorDacl(1, dacl, 0)
+    win32security.SetFileSecurity(file_name, win32security.DACL_SECURITY_INFORMATION, sd)
+
+@contextlib.contextmanager
+def openNoExe(file_name):
+    try:
+        f = open(file_name, 'wb')
+    except IOError, detail:
+        raise SCons.Errors.InternalError('Unable to open "' +
+                                         str(file_name) + '" for writing:' + str(detail))
+    try:
+        yield f
```

Comment:
I understand the point of context managers in general. 

But I don't actually see any real cleanup (did I miss something?) and if you do want to clean up, `try`/`finally` or `try`/`else` are often the way to go over a context manager.  Remember, you don't really need to close file handlers unless you intend to open the underlying file again immediately in that same program.  If you can just do `with f as open(...): do_this(f)` you of course should - but if you end up having to build your own context managers then it's really not worth it.